doc/design/dist/submit-job.md

```diff
@@ -0,0 +1,137 @@
+
+# PaddlePaddle Client
+PaddlePaddle client is command line tool, you can use a PaddlePaddle client to start a local train and submit a distributed training job to kubernetes cluster.
+
+The relation of PaddlePaddle, kubernetes and docker:
+
+<img src="./submit-job.png" width="500">
```

Comment:
> 1. I am not sure that pservers and trainers should be in two jobs

现在的配置会有两个问题：
1. 不利于trainer的大规模训练，由于trainer count==pserver count，当trainer数很大时，我们并不需要同样数量的pserver数，因为这会增加pserver失败的概率以及增加网络负载。
1. 由于启在同一个container里，pserver和trainer有一个会以后台方式启动，不符合container的设计原则，不利于故障检测和恢复。


> each trainer has a pserver running on the same physical node to optimally overlay networking and computing

看起来控制pserver的数量更可以达到优化网络的效果，而且trainer需要和所有的pserver通信，所以只有一个pserver 节点启在本地看起来效果也不是很大？

> paddle must communicate with Kubernetes' API server to start the job. It might communicate to the master process of the job too

我觉得 @helinwang 的这个[comment](https://github.com/PaddlePaddle/Paddle/pull/1770#discussion_r111271275)是有道理的，我paddle可以只和master进行通信，master作为一个service存在，并且这个master和 https://github.com/PaddlePaddle/Paddle/tree/develop/doc/design/dist#master-process 这里的master 并不是同一个master, 我会在下一版的更新中修改这部分的描述。